src/rust/cryptography-x509-verification/src/policy/mod.rs

```diff
@@ -234,18 +234,17 @@ pub struct Policy<'a, B: CryptoOps> {
 }
 
 impl<'a, B: CryptoOps> Policy<'a, B> {
-    /// Create a new policy with defaults for the certificate profile defined in
-    /// the CA/B Forum's Basic Requirements.
-    pub fn new(
+    fn new(
         ops: B,
-        subject: Subject<'a>,
+        subject: Option<Subject<'a>>,
         time: asn1::DateTime,
         max_chain_depth: Option<u8>,
     ) -> Self {
+        let has_subject = subject.is_some();
```

Comment:
NB: We need this binding because `subject` is stolen by the `Self { ... }` ctor, meaning that we can't bind on it in the `match` below.